truffleruby/src/main/java/org/truffleruby/cext/CExtNodes.java

```diff
@@ -250,6 +251,56 @@ public Object ulong2num(long num) {
 
     }
 
+    @CoreMethod(names = "rb_big_pack", isModuleFunction = true, required = 2)
+    public abstract static class BIGPACKNode extends CoreMethodArrayArgumentsNode {
+        @Specialization(guards = "isRubyBignum(num)")
+        public DynamicObject bigPack(DynamicObject num, long length) {
+            BigInteger bi = Layouts.BIGNUM.getValue(num);
+            long[] longs = new long[(int)length];
+            byte[] bytes = bi.toByteArray();
+            int limit = Math.min(longs.length, bytes.length / 8);
+            int fraction = limit == longs.length ? 0 : bytes.length % 8;
+            int blank_start = fraction == 0 ? limit : limit + 1;
+            boolean negative = bi.signum() == -1;
+            for (int i = 0; i < limit; i++) {
+                long x = 0;
+                for (int j = 0; j < 8; j++) {
+                    long a_byte = bytes[bytes.length - 1 - i * 8 - j] & 0xffl;
+                    x = x | (a_byte << (j * 8));
+                }
+                longs[i] = x;
+            }
+            if (fraction != 0) {
+                long x = 0;
+                for (int i = 0; i < fraction; i++) {
+                    long a_byte = bytes[bytes.length - 1 - limit * 8 - i] & 0xffl;
+                    x = x | (a_byte << (i * 8));
+                }
+                if (negative) {
+                    for (int i = fraction; i < 8; i++) {
+                        x = x | (0xffl << (i * 8));
+                    }
+                }
+                System.out.flush();
+                longs[limit] = x;
+            }
+            if (negative) {
+                for (int i = blank_start; i < length; i++) {
+                    longs[i] = -1;
+                }
+            }
```

Comment:
I've added a comment. This area is likely to get refactored soon anyway as it is a special case of rb_integer_pack.